symbolic-demangle/src/lib.rs

```diff
@@ -176,6 +176,21 @@ fn try_demangle_cpp(ident: &str, opts: DemangleOptions) -> Option<String> {
     {
         use cpp_demangle::{DemangleOptions as CppOptions, Symbol as CppSymbol};
 
+        // If ident has a suffix of $ followed by 32 hex digits, discard it.
+        let ident = {
+            let n = ident.len();
+            if n < 33 {
```

Comment:
Let's use a speaking variable name here. 

This entire block is a good candidate to be moved into a private function with a descriptive name.